java/code/src/com/redhat/rhn/frontend/servlets/SystemDetailsMessageFilter.java

```diff
@@ -57,10 +57,20 @@ public void doFilter(
         RequestContext rctx = new RequestContext(req);
         Long sid = rctx.getRequiredParam("sid");
         User user = rctx.getCurrentUser();
-        Server s  = SystemManager.lookupByIdAndUser(sid, user);
-        s.asMinionServer().ifPresentOrElse(
+        Server s = SystemManager.lookupByIdAndUser(sid, user);
+        this.processSystemMessages(req, s, sid);
+    }
+
+    /**
+     * Process a system and add the necessary warning messages for it if necessary
+     * @param req - the servlet request
+     * @param server - the system to be processed
+     * @param sid - the id of the system
+     */
+    public void processSystemMessages(HttpServletRequest req, Server server, Long sid) {
+        server.asMinionServer().ifPresentOrElse(
             minion -> processMinionMessages(req, minion),
-            () -> addMessageIfNecessary(req, true, TRADITIONAL_STACK_MESSAGE_KEY)
+            () -> addMessageIfNecessary(req, SystemManager.isTraditional(sid), TRADITIONAL_STACK_MESSAGE_KEY)
```

Comment:
We already have a function to check the entitlements of the system. I think you can use:
```
() -> addMessageIfNecessary(req, SystemManager.hasEntitlement(sid, EntitlementManager.MANAGEMENT), TRADITIONAL_STACK_MESSAGE_KEY)
```
instead

Reply:
@parlt91 thank you for the review. There are two problems with using `SystemManager.hasEntitlement` here:

1. It receives the entire entitlement object, not only the label, so it would be necessary to restore the object from the database using the label; and
2. It seems to be not optimized as necessary here, considering that it is listing all the entitlements from the database in order to compare with the one you provide as parameter.

So, if you agree I would prefer to keep the implementation as proposed. What do you think?

Reply:
Ok good point. From my side this is a satisfying answer. I am going to approve. Thank you

Reply:
Ok good point. From my side this is a satisfying answer. I am going to approve. Thank you

Reply:
There aren't that many entitlements on one server and this message is only on a page displaying server details. Performance impact is very low here.

Reply:
Ok, thank you. I changed the implementation iterate through the entitlement list.